controllers/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -91,6 +123,45 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
 
 	annotations := pvc.GetAnnotations()
 	schedule, scheduleFound := getScheduleFromAnnotation(&logger, annotations)
+	if !scheduleFound {
```

Comment:
This is quite a large if-block, with lots of (good!) notes. Did you consider making it a helper function?

Reply:
Reduced the code length in the block now, it should be good